ext/hal/st/stm32cube/stm32f1xx/drivers/include/stm32f1xx_hal_eth.h

```diff
@@ -2040,6 +2040,8 @@ void HAL_ETH_MspDeInit(ETH_HandleTypeDef *heth);
 HAL_StatusTypeDef HAL_ETH_DMATxDescListInit(ETH_HandleTypeDef *heth, ETH_DMADescTypeDef *DMATxDescTab, uint8_t *TxBuff, uint32_t TxBuffCount);
 HAL_StatusTypeDef HAL_ETH_DMARxDescListInit(ETH_HandleTypeDef *heth, ETH_DMADescTypeDef *DMARxDescTab, uint8_t *RxBuff, uint32_t RxBuffCount);
 
+HAL_StatusTypeDef HAL_ETH_CheckAutoNegotiation(ETH_HandleTypeDef *heth);
```

Comment:
You're not supposed to modify files in ext/hal/....
Whatever additional functionality you need should be implemented in drivers/ethernet/

Reply:
I've been looking for a way to implement the functionality solely under the driver part, not touching the ext/hal code. 

One way would be to partial or fully implement the HAL_ETH_Init(..) function in the driver/net/eth_stm32_hal.c codebase. Which doesn't speak for using the external low level drivers. 
Another way would be to call HAL_ETH_Init(..) every time the PHY state changes. There are two things why that wouldn't be ideal: 1) the init function needs to be called at boot anyhow to enable PHY communication and 2) more things than just the link status are being initialized. 
The init function has a long delay in case there's no link detected, which cannot be removed when leaving the ext/hal/.. code untouched and not reimplement it in the driver. 

I would like to ask the maintainers: @tbursztyka @erwango @pfalcon @jukkar what they would propose to overcome this long delay at boot.

Reply:
@vlotech: checking for alternate and acceptable solution. Coming back to you.